hack/gen-patch-defaultrestmapper.sh

```diff
@@ -0,0 +1,109 @@
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")"/.. && pwd)
+DEFAULTRESTMAPPER_FILEPATH="${DEFAULTRESTMAPPER_FILEPATH:-$( go list -m -json k8s.io/apimachinery | jq -r .Dir )/pkg/api/meta/restmapper.go}"
+DEFAULTRESTMAPPER_PATCH_FILEPATH="${DEFAULTRESTMAPPER_PATCH_FILEPATH:-${REPO_ROOT}/pkg/reconciler/dynamicrestmapper/defaultrestmapper_kcp.go}"
+
+function usage {
+    echo "gen-patch-defaultrestmapper copies upstream's DefaultRESTMapper,"
+    echo "makes the changes needed by our DynamicRESTMapper, and stores"
+    echo "the result in ${DEFAULTRESTMAPPER_PATCH_FILEPATH}."
+    echo
+    echo "By default, gen-patch-defaultrestmapper tries to look for"
+    echo "restmapper.go in:"
+    echo
+    printf "\t%s,\n" "${DEFAULTRESTMAPPER_FILEPATH}"
+    echo
+    echo "as detected by 'go list -m -json k8s.io/apimachinery'."
+    echo "If you wish to use a different path, pass it in through"
+    echo "DEFAULTRESTMAPPER_FILEPATH environment variable."
+    echo
+    echo "Usage:"
+    echo
+    printf "\t%s\n" "${0}"
+    exit 0
+}
+
+# Script input validation.
+
+[[ $# != 0 ]] && usage
+[[ -f "${DEFAULTRESTMAPPER_FILEPATH}" ]] || { echo "Error: ${DEFAULTRESTMAPPER_FILEPATH} (from DEFAULTRESTMAPPER_FILEPATH) is not a file" 1>&2 ; exit 1 ; }
+touch "${DEFAULTRESTMAPPER_PATCH_FILEPATH}"
+[[ -f "${DEFAULTRESTMAPPER_PATCH_FILEPATH}" ]] || { echo "Error: ${DEFAULTRESTMAPPER_PATCH_FILEPATH} (from DEFAULTRESTMAPPER_PATCH_FILEPATH) is not a file" 1>&2 ; exit 1 ; }
+
+# File generation.
+
+# First, we generate the preamble with our package name, and
+# paste upstream's defaultrestmapper.go contents, starting right
+# after its `package meta` declaration.
+
+cat "${REPO_ROOT}/hack/boilerplate/boilerplate.go.txt" > "${DEFAULTRESTMAPPER_PATCH_FILEPATH}"
+sed -i "s/YEAR/$(date +'%Y')/" "${DEFAULTRESTMAPPER_PATCH_FILEPATH}"
+cat >> "${DEFAULTRESTMAPPER_PATCH_FILEPATH}" << Header_EOF
+
+package dynamicrestmapper
+
+// Code generated by hack/gen-patch-defaultrestmapper.sh
+// Original file in k8s.io/apimachinery/pkg/api/meta/restmapper.go.
+
+// We need this DefaultRESTMapper fork to be able to modify its
+// internal mappings, to support update and deletion operations.
+// These are implemented separately, in defaultrestmapper_mutable.go.
+
+Header_EOF
+
+awk '/^package meta$/ {flag=1; next} flag' "${DEFAULTRESTMAPPER_FILEPATH}" >> "${DEFAULTRESTMAPPER_PATCH_FILEPATH}"
+
+# Next, we need to replace symbols that were local to the meta
+# package with their package-qualified names and fix the imports
+# in the resulting file.
+
+# This is the list of all symbols that need to be referenced inside
+# the k8s.io/apimachinery/pkg/api/meta package.
+#
+# Please keep it up-to-date and sorted.
+symbols_from_meta_pkg=(
+    AmbiguousResourceError
+    NoKindMatchError
+    NoResourceMatchError
+    ResettableRESTMapper
+    RESTMapper
+    RESTMapping
+    RESTScope
+    RESTScopeName
+    RESTScopeNameNamespace
+    RESTScopeNameRoot
+)
```

Comment:
Does anything produce a failure if this list isn't updated and new symbols come in?
I'd guess so since the symbol probably isn't available in the local package and would then produce a compile error, but wanna make sure.